static/src/javascripts/projects/common/modules/video/ga-helper.ts

```diff
@@ -0,0 +1,105 @@
+type EventType =
+	| 'play'
+	| 'skip'
+	| '25'
+	| '50'
+	| '75'
+	| 'watched25'
+	| 'watched50'
+	| 'watched75'
+	| 'end';
+
+type Metrics =
+	| 'metric1'
+	| 'metric2'
+	| 'metric3'
+	| 'metric4'
+	| 'metric5'
+	| 'metric6';
+
+type EventMetrics = {
+	play?: 'metric1';
+	skip?: 'metric2';
+	'25'?: 'metric3';
+	'50'?: 'metric4';
+	'75'?: 'metric5';
+	watched25?: 'metric3';
+	watched50?: 'metric4';
+	watched75?: 'metric5';
+	end?: 'metric6';
+};
+
+type MediaEvent = {
+	mediaId: string;
+	mediaType: string;
+	eventType: EventType;
+	isPreroll: boolean;
+};
+
+type InitialFieldsObject = {
+	eventCategory: string;
+	eventAction: string;
+	eventLabel: string;
+	dimension19: string;
+	dimension20: string;
+};
+type MetricFieldsObject = {
+	[key in Metrics]?: 1;
+};
```

Comment:
Similar to above, if you want to use utility types.

```suggestion
type MetricFieldsObject = Record<Metrics, 1>
```